rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf

```diff
@@ -167,7 +166,7 @@ SecRule REQUEST_HEADERS:Content-Length "!@rx ^\d+$" \
 # empty, then it will match.
 #
 # -=[ References ]=-
-# https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.3
+# https://www.rfc-editor.org/rfc/rfc9110.html
```

Comment:
Same.

Reply:
Maybe https://www.rfc-editor.org/rfc/rfc9110.html#section-6.4?

Reply:
Well...looks like I need to read the standard all over again. Has changed a lot.

Reply:
Saw an interesting thread: https://stackoverflow.com/questions/3854842/content-length-header-with-head-requests

Also In [#8.6.4](https://www.rfc-editor.org/rfc/rfc9110.html#section-8.6-4) mentions about user agent:
> A user agent SHOULD send Content-Length in a request when the method defines a meaning for enclosed content and it is not sending Transfer-Encoding. For example, a user agent normally sends Content-Length in a POST request even when the value is 0 (indicating empty content). A user agent SHOULD NOT send a Content-Length header field when the request message does not contain content and the method semantics do not anticipate such data.

Reply:
Probably these two:

In Section https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.1:

> Although request message framing is independent of the method used, **content received in a GET request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection because of its potential as a request smuggling attack** ([Section 11.2](https://www.rfc-editor.org/rfc/rfc9112#section-11.2) of [[HTTP/1.1](https://www.rfc-editor.org/rfc/rfc9110.html#HTTP11)]). A client SHOULD NOT generate content in a GET request unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported. An origin server SHOULD NOT rely on private agreements to receive content, since participants in HTTP communication are often unaware of intermediaries along the request chain.

and https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.2:
> Although request message framing is independent of the method used, **content received in a HEAD request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection because of its potential as a request smuggling attack** ([Section 11.2](https://www.rfc-editor.org/rfc/rfc9112#section-11.2) of [[HTTP/1.1](https://www.rfc-editor.org/rfc/rfc9110.html#HTTP11)]).

Reply:
Yes, I think both 9.3.1 and 9.3.2 indicates that:

>  `A client SHOULD NOT generate content in a GET request` unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported.

> `A client SHOULD NOT generate content in a HEAD request` unless it is made directly to an origin server that has previously indicated, in or out of band, that such a request has a purpose and will be adequately supported.

updated in [423f6ef](https://github.com/coreruleset/coreruleset/pull/3232/commits/423f6ef60134ac8ff173baa0c2be85cad83f71f0)

Reply:
Thanks.